src/Function/Bundles/Related.agda

```diff
@@ -0,0 +1,290 @@
+------------------------------------------------------------------------
+
+-- The Agda standard library
+--
+-- Relatedness for the function hierarchy
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Bundles.Related where
+
+open import Level
+open import Relation.Binary using (Sym; Reflexive; Trans; IsEquivalence; Setoid; IsPreorder; Preorder)
+import Function.Related as R
+open import Function.Bundles
+open import Function.Base
+open import Function.Equality using (_⟨$⟩_)
+import Function.Equivalence as Eq
+import Function.Injection   as Inj
+import Function.Inverse     as Inv
+import Function.LeftInverse as LeftInv
+import Function.Surjection  as Surj
+import Relation.Binary.PropositionalEquality as P
+
+open import Function.Properties.Inverse
+import Function.Construct.Symmetry as Symmetry
+import Function.Construct.Identity as Identity
+import Function.Construct.Composition as Composition
+
+open import Data.Product
+
+infix 4 _∼[_]_
+
+_∼[_]_ : ∀ {ℓ₁ ℓ₂} → Set ℓ₁ → R.Kind → Set ℓ₂ → Set _
```

Comment:
As this file is going to be the one that we recommend using going forwards, then I think we should move all the shared definitions, (e.g. `Kind`) to _this_ file and then re-export them publicly from `Function.Related` to maintain backwards compatibility.

Reply:
Makes sense, I'll do so!